src/axom/quest/examples/quest_marching_cubes_example.cpp

```diff
@@ -0,0 +1,1019 @@
+// Copyright (c) 2017-2022, Lawrence Livermore National Security, LLC and
+// other Axom Project Developers. See the top-level COPYRIGHT file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+/*!
+ \file marching_cubes_example.cpp
+ \brief Driver and test for a marching cubes iso-surface generation
+
+  The test can generate planar and round contours.  Planar contours
+  can be checked to machine-zero accuracy, but it doesn't test a great
+  variety of contour-mesh intersection types.  Round contours can
+  check more intersection types but requires a tolerance to allow
+  for the function not varying linearly along mesh lines.
+*/
+
+// Axom includes
+#include "axom/config.hpp"
+#include "axom/core.hpp"
+#include "axom/slic.hpp"
+#include "axom/primal.hpp"
+#include "axom/mint.hpp"
+#include "axom/quest.hpp"
+#include "axom/sidre.hpp"
+#include "axom/core/Types.hpp"
+
+#include "conduit_blueprint.hpp"
+#include "conduit_relay_io_blueprint.hpp"
+#ifdef AXOM_USE_MPI
+  #include "conduit_blueprint_mpi.hpp"
+  #include "conduit_relay_mpi_io_blueprint.hpp"
+#endif
+
+#include "axom/fmt.hpp"
+#include "axom/CLI11.hpp"
+
+#ifdef AXOM_USE_MPI
+  #include "mpi.h"
+#endif
+
+// C/C++ includes
+#include <string>
+#include <limits>
+#include <map>
+#include <vector>
+#include <cmath>
+
+#ifndef __WHERE
+  #define __STRINGIZE(x) __STRINGIZE2(x)
+  #define __STRINGIZE2(x) #x
+  //!@brief String literal for code location
+  #define __WHERE \
+    __FILE__ ":" __STRINGIZE(__LINE__) "(" + std::string(__func__) + ") "
+#endif
+
+namespace quest = axom::quest;
+namespace slic = axom::slic;
+namespace mint = axom::mint;
+namespace sidre = axom::sidre;
+namespace primal = axom::primal;
+namespace mint = axom::mint;
+namespace numerics = axom::numerics;
+
+///////////////////////////////////////////////////////////////
+// converts the input string into an 80 character string
+// padded on both sides with '=' symbols
+std::string banner(const std::string& str)
+{
+  return axom::fmt::format("{:=^80}", str);
+}
+
+///////////////////////////////////////////////////////////////
+/// Struct to parse and store the input parameters
+struct Input
+{
+public:
+  std::string meshFile;
+  std::string fieldsFile {"fields"};
+
+  // Center of round contour function
+  bool usingRound {false};
+  std::vector<double> fcnCenter;
+
+  // Parameters for planar contour function
+  bool usingPlanar {false};
+  std::vector<double> inPlane;
+  std::vector<double> perpDir;
```

Comment:
Note: Primal has a `Plane` class which has a `signedDistance()` function

Reply:
I'll check it out.